pkg/controller/cloud/pvlcontroller.go

```diff
@@ -192,105 +182,22 @@ func (pvlc *PersistentVolumeLabelController) addLabels(key string) error {
 
 func (pvlc *PersistentVolumeLabelController) addLabelsToVolume(vol *v1.PersistentVolume) error {
 	var volumeLabels map[string]string
-
 	// Only add labels if in the list of initializers
 	if needsInitialization(vol.Initializers, initializerName) {
-		if vol.Spec.AWSElasticBlockStore != nil {
-			labels, err := pvlc.findAWSEBSLabels(vol)
+		if labeler, ok := (pvlc.cloud).(cloudprovider.PVLabeler); ok {
+			labels, err := labeler.GetLabelsForVolume(vol)
 			if err != nil {
-				return fmt.Errorf("error querying AWS EBS volume %s: %v", vol.Spec.AWSElasticBlockStore.VolumeID, err)
+				return fmt.Errorf("error querying volume %v: %v", vol.Spec, err)
 			}
 			volumeLabels = labels
+			return pvlc.updateVolume(vol, volumeLabels)
```

Comment:
Need to do the update regardless of whether the cloud provider supports labeling so that the initializer will be removed.  Move the update to outside the labeler, ok if block above.

Reply:
ah i missed that. will do. thanks